docs/spec_draft.md

```diff
@@ -927,6 +928,47 @@ spaces of `result` and `operand`.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.reverse
+
+### Semantics
+
+Reverses the order of elements in the `operand` along the specified `dimensions`, generating an output array of the same shape. Each element of the operand array at a multidimensional index is stored into the output array at a transformed index. The multidimensional index is transformed by reversing the index in each dimension to be reversed (i.e., if a dimension of size N is one of the reversing dimensions, its index i is transformed into N - 1 - i).
+
+### Operands
+
+| Name         | Type                               |
+|--------------|------------------------------------|
+| `operand`    | tensor of any supported types      |
+| `dimensions` | 1-dimensional array of type `si64` |
+
+### Results
+
+| Name     | Type                          |
+|----------|-------------------------------|
+| `result` | tensor of any supported types |
+
+### Constraints
+
+  * (C1) `operand` and `result` have the same element type.
```

Comment:
Additionally:

```
* (C3) All dimensions in `dimensions` are unique.
* (C4) For all dimensions in `dimensions`, 0 $\le$ `dimension` $\lt$ `R`, where `R` is the rank of the `result`.
```

The checklist mentioned above provides links to files which contain existing implementations of constraints for HLO and StableHLO. It's a good idea to derive constraints from there.